vetomint/tests/integration_test.rs

```diff
@@ -83,3 +85,594 @@ fn success_trivial_1() {
         vec![ConsensusResponse::FinalizeBlock { proposal: 0 }]
     );
 }
+
+#[ignore]
+#[test]
+//New block with 2/3+ e.t., ununiformed voting-power and pre-committed
+fn new_block_et_two_thirds_precommitted2() {
+    let height_info = HeightInfo {
+        validators: vec![10, 8, 6, 5, 4, 2, 2],
+        this_node_index: 3,
+        timestamp: 0,
+        consensus_params: ConsensusParams { timeout_ms: 1000 },
+    };
+    let mut state = ConsensusState::new(height_info.clone());
+
+    // STEP 1: Proposal.
+    let event = ConsensusEvent::BlockProposal {
+        proposal: 0,
+        proposer: 0,
+        round: 0,
+        time: 1,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert!(response.is_empty());
+    let event = ConsensusEvent::ProposalFavor {
+        proposal: 0,
+        favor: true,
+        time: 2,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert_eq!(
+        response,
+        vec![ConsensusResponse::BroadcastPrevote {
+            proposal: 0,
+            round: 0
+        }]
+    );
+
+    // STEP 2: Prevote.
+    for validator_index in 0..=1 {
+        let event = ConsensusEvent::Prevote {
+            proposal: 0,
+            round: 0,
+            signer: validator_index,
+            time: 3,
+        };
+        let response = progress(&height_info, &mut state, event);
+        assert!(response.is_empty());
+    }
+    let event = ConsensusEvent::Prevote {
+        proposal: 0,
+        round: 0,
+        signer: 2,
+        time: 3,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert_eq!(
+        response,
+        vec![ConsensusResponse::BroadcastPrecommit {
+            proposal: 0,
+            round: 0
+        }]
+    );
+
+    // STEP 3: Prevote.
+    for validator_index in 0..=1 {
+        let event = ConsensusEvent::Precommit {
+            proposal: 0,
+            round: 0,
+            signer: validator_index,
+            time: 4,
+        };
+        let response = progress(&height_info, &mut state, event);
+        assert!(response.is_empty());
+    }
+    let event = ConsensusEvent::Precommit {
+        proposal: 0,
+        round: 0,
+        signer: 2,
+        time: 4,
+    };
+    let response = progress(&height_info, &mut state, event);
+    assert_eq!(
+        response,
+        vec![ConsensusResponse::FinalizeBlock { proposal: 0 }]
+    );
+}
+
+#[ignore]
+#[test]
+//New block with 5/6+ e.t. and pre-committed
+fn new_block_et_five_sixths_precommitted() {
```

Comment:
This test scenario is not about 5/6+ e.t.

Reply:
nodes,0, 1, 2, 3, 6, prevote to proposal and node 4 prevote to nil
total 6 of 7 nodes prevote, thus I think it is about 5/6+ e.t.
can you describe with more detail?